File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys
input = sys.stdin.readline

queue = deque()

for _ in range(int(input())):
    cmd = input().split()

    if cmd[0] == "push_front":
        queue.appendleft(cmd[1])

    elif cmd[0] == "push_back":
        queue.append(cmd[1])

    elif cmd[0] == "pop_front":
        if queue:
            print(queue[0])    
            queue.popleft()
        else:
            print("-1")

    elif cmd[0] == "pop_back":
        if queue:
            print(queue[len(queue) - 1])    
            queue.pop()
        else:
            print("-1")

    elif cmd[0] == "size":
        print(len(queue))

    elif cmd[0] == "empty":
        if len(queue) == 0: 
            print(1)
        else: 
            print(0)

    elif cmd[0] == "front":
        if len(queue) == 0: 
            print(-1)
        else: 
            print(queue[0])

    elif cmd[0] == "back":
        if len(queue) == 0: 
            print(-1)
        else: 
            print(queue[-1])

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.Deque;
import java.util.LinkedList;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> dq = new LinkedList<>();

        while (n-- > 0) {
            String[] cmd = br.readLine().split(" ");
            if (cmd[0].equals("push_front")) {
                dq.offerFirst(Integer.parseInt(cmd[1]));
            } else if (cmd[0].equals("push_back")) {
                dq.offerLast(Integer.parseInt(cmd[1]));
            } else if (cmd[0].equals("pop_front")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.pollFirst() + "\n");
            } else if (cmd[0].equals("pop_back")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.pollLast() + "\n");
            } else if (cmd[0].equals("size")) {
                bw.write(dq.size() + "\n");
            } else if (cmd[0].equals("empty")) {
                bw.write(dq.isEmpty() ? "1\n" : "0\n");
            } else if (cmd[0].equals("front")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.getFirst() + "\n");
            } else if (cmd[0].equals("back")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.getLast() + "\n");
            }
        }
        bw.close();
    }
}
